server.js

```diff
@@ -114,10 +114,14 @@ const collectDurationPromise = (name, func) => {
   }
 }
 collectDurationPromise('validatePreconditions', require('./lib/startup/validatePreconditions'))()
-collectDurationPromise('restoreOverwrittenFilesWithOriginals', require('./lib/startup/restoreOverwrittenFilesWithOriginals'))()
 collectDurationPromise('cleanupFtpFolder', require('./lib/startup/cleanupFtpFolder'))()
 collectDurationPromise('validateConfig', require('./lib/startup/validateConfig'))()
 
+// Reloads the i18n files in case of server restarts or starts.
+async function reloadI18nFiles () {
+  collectDurationPromise('restoreOverwrittenFilesWithOriginals', require('./lib/startup/restoreOverwrittenFilesWithOriginals'))()
```

Comment:
This needs to be awaited. Otherwise we'd still have the race condition

```suggestion
  await collectDurationPromise('restoreOverwrittenFilesWithOriginals', require('./lib/startup/restoreOverwrittenFilesWithOriginals'))()
```